src/test/java/it/gov/pagopa/pu/classification/service/BalanceTemplateResolverServiceTest.java

```diff
@@ -0,0 +1,167 @@
+    ctAccertamentoDefault.setImporto("TOTALE");
+    ctCapitoloDefault.getAccertamento().add(ctAccertamentoDefault);
+    ctBilancioDefault.getCapitolo().add(ctCapitoloDefault);
+
+    String balanceExpected = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?><bilancio xmlns=\"http://www.regione.veneto.it/schemas/2012/Pagamenti/Ente/\"><capitolo><codCapitolo>CAP1</codCapitolo><accertamento><importo>100.00</importo></accertamento></capitolo></bilancio>";
+
+    when(balanceServiceMock.unmarshalBalance(request.getBalance())).thenReturn(ctBilancioDefault);
+    String result = service.calculateAmountBalance(request);
+
+    assertEquals(balanceExpected, result);
+    assertDoesNotThrow(() -> balanceUnmarshallerService.unmarshal(result));
+  }
+
+  @Test
+  void givenBalanceDefaultWhenCalculateThenReturnCalculatedBalance() {
+    String balance = "<bilancio><capitolo><codCapitolo>CAP1</codCapitolo><accertamento>" +
+      "<importo>function calcola_importo(IMPORTO) {var importo = Number(IMPORTO);var risultato = importo * 0.80;return (Math.ceil(risultato * 100) / 100).toString();}</importo>" +
+      "</accertamento></capitolo></bilancio>";
+    CalculateAmountBalanceRequest request = CalculateAmountBalanceRequest.builder()
+      .balance(balance)
+      .amountCents(10000L)
+      .remittanceInformation("remittanceInformation")
+      .build();
+
+    CtBilancioDefault ctBilancioDefault = new CtBilancioDefault();
+    CtCapitoloDefault ctCapitoloDefault = new CtCapitoloDefault();
+    ctCapitoloDefault.setCodCapitolo("CAP1");
+    CtAccertamentoDefault ctAccertamentoDefault = new CtAccertamentoDefault();
+    ctAccertamentoDefault.setImporto("function calcola_importo(IMPORTO){var importo = Number(IMPORTO);var risultato = importo * 0.80;return (Math.ceil(risultato * 100) / 100).toString();}");
+    ctCapitoloDefault.getAccertamento().add(ctAccertamentoDefault);
+    ctBilancioDefault.getCapitolo().add(ctCapitoloDefault);
+
+    String balanceExpected = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?><bilancio xmlns=\"http://www.regione.veneto.it/schemas/2012/Pagamenti/Ente/\"><capitolo><codCapitolo>CAP1</codCapitolo><accertamento><importo>80.00</importo></accertamento></capitolo></bilancio>";
+
+    when(balanceServiceMock.unmarshalBalance(request.getBalance())).thenReturn(ctBilancioDefault);
+
+    String result = service.calculateAmountBalance(request);
+
+    assertEquals(balanceExpected, result);
+    assertDoesNotThrow(() -> balanceUnmarshallerService.unmarshal(result));
+  }
+
+  @Test
+  void givenBalanceDefaultWhenCalculateExtractThenReturnCalculatedBalance() {
+    String balance = "<bilancio><capitolo><codCapitolo>CAP1</codCapitolo><accertamento><importo>function estrai_importo(IMPORTO) { return '90.00'; }</importo></accertamento></capitolo></bilancio>";
+    CalculateAmountBalanceRequest request = CalculateAmountBalanceRequest.builder()
+      .balance(balance)
+      .amountCents(10000L)
```

Comment:
This way is more readable to understand it as euro
```suggestion
      .amountCents(100_00L)
```